mavros_extras/src/plugins/odom.cpp

```diff
@@ -31,103 +31,38 @@ using Matrix6d = Eigen::Matrix<double, 6, 6, Eigen::RowMajor>;
 /**
  * @brief Odometry plugin
  *
- * Sends odometry data to the FCU position and attitude estimators.
- * @see odom_cb()
+ * Sends odometry data to the FCU estimator and
+ * publishes odometry data that comes from FCU.
+ *
+ * This plugin is following ROS REP 147. Pose is expressed in parent frame. (Quaternion rotates from child to parent)
+ * The twist is expressed in the child frame.
+ *
+ * @see odom_cb()	transforming and sending odometry to fcu
+ * @see handle_odom()	receiving and transforming odometry from fcu
  */
 class OdometryPlugin : public plugin::PluginBase {
 public:
-	EIGEN_MAKE_ALIGNED_OPERATOR_NEW 	// XXX(vooon): added to try to fix #1223. Not sure that it is needed because class do not have Eigen:: fields.
+	EIGEN_MAKE_ALIGNED_OPERATOR_NEW		// XXX(vooon): added to try to fix #1223. Not sure that it is needed because class do not have Eigen:: fields.
 
 	OdometryPlugin() : PluginBase(),
 		odom_nh("~odometry"),
-		body_frame_orientation_in_desired("flu"),
-		body_frame_orientation_out_desired("frd"),
-		child_frame_id("base_link"),
-		frame_id("odom"),
-		local_frame_in("local_origin_ned"),
-		local_frame_out("vision_ned")
+		fcu_odom_parent_id_des("map"),
+		fcu_odom_child_id_des("base_link")
 	{ }
 
 	void initialize(UAS &uas_)
 	{
 		PluginBase::initialize(uas_);
 
-		odom_nh.param<std::string>("frame_id", frame_id, "odom");
-		odom_nh.param<std::string>("child_frame_id", child_frame_id, "base_link");
-
-		// frame tf params:
-		odom_nh.param<std::string>("in/frame_tf/local_frame", local_frame_in, "local_origin_ned");
-		odom_nh.param<std::string>("out/frame_tf/local_frame", local_frame_out, "vision_ned");
-		odom_nh.param<std::string>("in/frame_tf/body_frame_orientation", body_frame_orientation_in_desired, "flu");
-		odom_nh.param<std::string>("out/frame_tf/body_frame_orientation", body_frame_orientation_out_desired, "frd");
-
-		boost::algorithm::to_lower(local_frame_out);
-		boost::algorithm::to_lower(body_frame_orientation_out_desired);
+		// frame params:
+		odom_nh.param<std::string>("fcu/odom_parent_id_des", fcu_odom_parent_id_des, "map");
+		odom_nh.param<std::string>("fcu/odom_child_id_des", fcu_odom_child_id_des, "base_link");
```

Comment:
Note to self: these are the ROS frames the mavlink ODOMETRY data from the FCU is published in.